addons/purchase_stock/models/stock.py

```diff
@@ -19,6 +19,42 @@ class StockMove(models.Model):
     created_purchase_line_id = fields.Many2one('purchase.order.line',
         'Created Purchase Order Line', ondelete='set null', readonly=True, copy=False)
 
+    def _action_cancel_origin(self):
+        """ Update purchase order in case of draft state.
+            > Check if all previouse moves cancelled , then cancel or unlink purchase line.
+            > Check if some of previouse moves are not cancel decrease the line quantity.
+            > Check that purchase line has been generated from the multiple move or not.
+        """
+        moves_purchase_line = self.filtered(lambda m: m.created_purchase_line_id and m.state not in ('done', 'cancel'))
```

Comment:
why don't you directly override _action_cancel ?

Reply:
we use _action_cancel_origin, because, in that method, there is a mechanism, with which we easily get to know that the previous moves of the documents has been canceled, and now it is the time to make changes to the record, if record is in draft then system will update the record, and if it is not then system will generate activity for user, so that user can manually change the record.